src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -4537,53 +4166,42 @@ def get_service_connector_client(
 
     def list_service_connector_resources(
         self,
-        user_name_or_id: Union[str, UUID],
         workspace_name_or_id: Union[str, UUID],
         connector_type: Optional[str] = None,
         resource_type: Optional[str] = None,
         resource_id: Optional[str] = None,
+        filter_model: Optional[ServiceConnectorFilter] = None,
```

Comment:
This will break the Liskov Substitution Principle because the interface signature is not the same. Maybe you should just add the filter to all methods, including the REST store and Client. I also think you can get rid of the other arguments and only rely on what's set in the filter. The exception is the `resource_id`, which is applied separately and not as part of the filter when you call `list_service_connectors`.

Reply:
Are you sure it breaks it? It seems to me that adding an additional argument **with default value** to a method of a subclass would not break it, as the subclass can still be used in all places where the superclass can be used. `mypy` seems to agree and only complain if the argument does not have a default value.

I'm a little afraid of adding the filter model also for the rest case, as we would now send that as part of the body instead of query parameters which means potential changes for the frontend would be required?

Reply:
Fine by me if mypy isn't complaining.